src/charm.py

```diff
@@ -80,85 +86,41 @@ def set_pod_spec(self, event):
                 f"--mysql_config_user={mysql['user']}",
                 f"--mysql_config_password={mysql['password']}",
             ]
-            volumes = []
         except (IndexError, KeyError):
             db_args = ["--metadata_store_server_config_file=/config/config.proto"]
             config_proto = (
                 'connection_config: {sqlite: {filename_uri: "file:/data/mlmd.db"}}'
             )
-            volumes = [
-                {
-                    "name": "config",
-                    "mountPath": "/config",
-                    "files": [{"path": "config.proto", "content": config_proto}],
-                }
-            ]
 
-        config = self.model.config
+            container.push("/config/config.proto", config_proto, make_dirs=True)
 
         args = db_args + [
-            f"--grpc_port={config['port']}",
+            f"--grpc_port={self.model.config['port']}",
             "--enable_database_upgrade=true",
         ]
 
-        self.model.unit.status = MaintenanceStatus("Setting pod spec")
-        self.model.pod.set_spec(
-            {
-                "version": 3,
-                "containers": [
-                    {
-                        "name": "mlmd",
-                        "command": ["/bin/metadata_store_server"],
-                        "args": args,
-                        "imageDetails": image_details,
-                        "ports": [
-                            {
-                                "name": "grpc-api",
-                                "containerPort": int(self.model.config["port"]),
-                            },
-                        ],
-                        "volumeConfig": volumes,
-                        "kubernetes": {
-                            "livenessProbe": {
-                                "tcpSocket": {"port": "grpc-api"},
-                                "initialDelaySeconds": 3,
-                                "periodSeconds": 5,
-                                "timeoutSeconds": 2,
-                            },
-                            "readinessProbe": {
-                                "tcpSocket": {"port": "grpc-api"},
-                                "initialDelaySeconds": 3,
-                                "periodSeconds": 5,
-                                "timeoutSeconds": 2,
-                            },
-                        },
-                    }
-                ],
+        layer = {
+            "summary": "mlmd-operator",
+            "description": "mlmd-operator layer",
+            "services": {
+                "mlmd": {
+                    "override": "replace",
+                    "summary": "mlmd",
+                    "command": shlex.join(["/bin/metadata_store_server"] + args),
```

Comment:
Pebble needs a command-line, so join arguments with shlex.join() in a single string.